sklearn/model_selection/_validation.py

```diff
@@ -389,12 +394,18 @@ def cross_val_predict(estimator, X, y=None, labels=None, cv=None, n_jobs=1,
     X, y, labels = indexable(X, y, labels)
 
     cv = check_cv(cv, y, classifier=is_classifier(estimator))
+
+    # Ensure the estimator has implemented the passed decision function
+    if not hasattr(estimator, apply_func):
```

Comment:
@rvraghav93 callable is equivalent to hasattr(var, "__call__"), how can it be used here? Plus it's not supported in Python 3